### DISCUSSION TITLE
Terraform syntax highlighting appears to be broken on Github with nested parentheses and strings

### DISCUSSION THREAD

**Discussion by owattley-rotageek** [github-linguist/linguist#7122] [Classification]
Looks like its do do with string interpolation `"${var.foo}"` (which does work) when nested inside parentheses `[]`

Example:

I can't give the full source file because it's a private repo but here's a screenshot of how it's broken (line 246):

![image](https://github.com/user-attachments/assets/14c8c2e9-5d73-4c2e-9069-a9aa9e50a151)

Minimal broken example is probably:

```terraform
resource "some_resource" "foo" {
  some_prop = var.bar["${var.baz}"].other_prop
}
```

This works (no nested `${}` inside `""`)
```terraform
resource "some_resource" "foo" {
  some_prop = var.bar["baz"].other_prop
}
```

... but this works (nested `${}` inside `""` but no outer `[]`)

```terraform
resource "some_resource" "foo" {
  some_prop = "${var.baz}"
}
```

Example of a working syntax highlighter (VSCode):

![image](https://github.com/user-attachments/assets/472566f0-5d7b-4319-9591-21a4a1d85045)

**Comment by lildude** [comment]
Please see the docs: [There's a problem with the syntax highlighting of a file](https://github.com/github-linguist/linguist/blob/main/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file).

Closing as this is not a Linguist issue.

**Reply by owattley-rotageek** [reply]
Apologies, will repost in the appropriate place, thanks.
